scripts/generate_code_snippet.py

```diff
@@ -0,0 +1,27 @@
+"""Provides functions for generating code examples from a Python file."""
+import os
+
+def generate_code_examples(file_path, output_dir):
+    """Generate code examples from a Python file.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the Python file.
+    output_dir : str
+        Path to the output directory.
+
+    """
+    with open(file_path, "r") as file:
+        content = file.read()
+
+    examples = content.split("# ----------------------------------------------------------------")
```

Comment:
I'd suggest some other comment....

```py
# BREAK BLOCK
```

The problem with "dashes" is that pepole tend to do those by hand and end up placing an "irregular" number of dashes. In those cases, your algorithm won't catch the block break conditions.